ui/desktop/app/routes/application.js

```diff
@@ -35,7 +37,16 @@ export default class ApplicationRoute extends Route {
     await this.session.setup();
     const theme = this.session.get('data.theme');
     this.toggleTheme(theme);
-    return this.clusterUrl.updateClusterUrl();
+    await this.clusterUrl.updateClusterUrl();
+
+    // Setup the DB from a successful authentication restoration
+    if (this.session.isAuthenticated) {
+      const userId = this.session.data?.authenticated?.user_id;
+      const clusterUrl = this.clusterUrl.rendererClusterUrl;
+      if (userId && clusterUrl) {
+        this.indexedDb.setup(formatDbName(userId, clusterUrl));
```

Comment:
non-blocking question for understanding: why is the `indexedDb` setup method being called here when the session service `handleAuthentication` method already calls the `setup` method? Another question, why is the session service put in the `ui/desktop` folder and not the `addons/api` folder ?

Reply:
Great questions.

The session service is in the `ui/desktop` because it has to live in the application portion of the code. Depending if it's in admin UI or desktop, we will have slightly different logic (namely the URL to use). Otherwise the `api` addon would have to be aware of its clients. I'm actually still even debating whether some of these services should be living in `core` or `api`.

I'm requiring the database name to need the user's ID (for uniqueness) which means they have to be logged in, otherwise I could setup up on loading the ember app. The setup method is called "twice" because of two different scenarios: the one where the user needs to authenticate which means the one in `application.js` wasn't called bc there wasn't a user yet so didn't do anything, and the one where the user was already logged in and we're restoring their authentication which gets called in `application.js`. The key thing to note is `handleAuthentication` isn't called on auth restoration.